main.go

```diff
@@ -91,26 +89,34 @@ func main() {
 		os.Exit(1)
 	}
 
-	// Create a new aws session
-	var sess *session.Session
-	if localAWSEndpoint == "" {
-		// Production mode
-		sess = session.Must(session.NewSessionWithOptions(session.Options{
-			SharedConfigState: session.SharedConfigEnable,
-		}))
+	// If using a local endpoint we create a custom endpoint resolver, hard code
+	// the region and test credentials.
+	// Otherwise we leave the config loading to the default aws order (env vars,
+	// EC2 IMDS, etc.)
+	awsLoadConfigOpts := []func(*config.LoadOptions) error{}
+	if localAWSEndpoint != "" {
+		setupLog.Info("Using custom AWS Endpoint", "endpoint", localAWSEndpoint)
+		awsEndpointsResolver := aws.EndpointResolverWithOptionsFunc(func(service, region string, options ...interface{}) (aws.Endpoint, error) {
+			return aws.Endpoint{PartitionID: "aws", URL: localAWSEndpoint, SigningRegion: "eu-west-1"}, nil
+		})
+		awsLoadConfigOpts = append(
+			awsLoadConfigOpts,
+			config.WithEndpointResolverWithOptions(awsEndpointsResolver),
+			config.WithCredentialsProvider(credentials.NewStaticCredentialsProvider("test", "test", "test")),
+			config.WithRegion("eu-west-1"),
+		)
 	} else {
-		// Testing mode
-		sess = session.Must(session.NewSessionWithOptions(session.Options{
-			Config: aws.Config{
-				Region:           aws.String("eu-west-1"),
-				Credentials:      credentials.NewStaticCredentials("test", "test", ""),
-				S3ForcePathStyle: aws.Bool(true),
-				Endpoint:         aws.String(localAWSEndpoint),
-			},
-			SharedConfigState: session.SharedConfigEnable,
-		}))
+		setupLog.Info("Using default AWS endpoint")
+	}
+
+	// Load the config with the given options and create a new KMSCA
+	cfg, err := config.LoadDefaultConfig(context.Background(), awsLoadConfigOpts...)
+	if err != nil {
+		setupLog.Error(err, "Error loading default aws config")
+		os.Exit(1)
 	}
-	ca := kmsca.NewKMSCA(sess)
+	setupLog.Info("Using region", "region", cfg.Region)
```

Comment:
Should this be just 
```suggestion
	setupLog.Info("Using region", cfg.Region)
```
or do you need a string, and then a key-value pair?

Reply:
>`func (logr.Logger).Info(msg string, keysAndValues ...interface{})`
Info logs a non-error message with the given key/value pairs as context.
>
>The msg argument should be used to add some constant description to the log line. The key/value pairs can then be used to add additional variable information. The key/value pairs must alternate string keys and arbitrary values.

I think it is required as pairs.